pyproject.toml

```diff
@@ -168,26 +168,24 @@ module = [
 # ref: https://mypy.readthedocs.io/en/stable/existing_code.html#introduce-stricter-options
 [[tool.mypy.overrides]]
 # Start off with these
-warn_unused_configs = true
-warn_redundant_casts = true
```

Comment:
This is the semantic change here (the rest is unintentional autoformatting) — mypy warns that these are not module-specific. So if we want to use these, we should promote to the general mypy config.

This makes sense to me for `warn_unused_configs`; less so for `warn_redundant_casts` — does it make sense to you @Illviljan ?

Reply:
We actually already have `warn_redundant_casts` at the project level.

I've added `warn_unused_configs` at the project level too.

Reply:
These are a copy/paste job to simulate strict mypy for NamedArray, since it's not possible to specify --strict for certain files... 
If there's a better way to simulate strict then I'm all for it.